File: python/Remove Duplicates from Sorted List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        if head is None:
            return None
        
        current = head
        cursor = head.next
        while cursor:
            if cursor.val == current.val:
                current.next = None
                current.next = cursor.next
                cursor = current
            else:
                current = cursor
            cursor = cursor.next
                
        return head

File: cpp/Remove Duplicates from Sorted List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode *cursor = head;
        while (cursor && cursor->next) {
            if (cursor->val == cursor->next->val) {
                ListNode *tmp = cursor->next;
                cursor->next = cursor->next->next;
                delete tmp;
            } else {
                cursor = cursor->next;
            }
        }
        return head;
    }
};
